src/hatchet/utils/run.py

```diff
@@ -101,33 +113,106 @@ def main(args=None):
             ] + extra_args
         )
 
-    # ----------------------------------------------------
-
-    if config.run.combine_counts:
-        _stdout = sys.stdout
-        sys.stdout = StringIO()
-
-        combine_counts(args=[
-            '-c', f'{output}/rdr/normal.1bed',
-            '-C', f'{output}/rdr/tumor.1bed',
-            '-B', f'{output}/baf/tumor.1bed',
-            '-t', f'{output}/rdr/total.tsv'
-        ])
 
-        out = sys.stdout.getvalue()
-        sys.stdout.close()
-        sys.stdout = _stdout
-
-        os.makedirs(f'{output}/bb', exist_ok=True)
-        with open(f'{output}/bb/bulk.bb', 'w') as f:
-            f.write(out)
-
-    # ----------------------------------------------------
+    if config.run.fixed_width is None or not config.run.fixed_width:
+        # ----------------------------------------------------
+        # ----------------------------------------------------
+        # Variable-width/adaptive binning
+        
+        if config.run.count_reads:
+            os.makedirs(f'{output}/rdr', exist_ok=True)
+            count_reads(
+                args=[
+                    '-N', config.run.normal,
+                    '-T'
+                ] + config.run.bams.split() + [
+                    '-S'
+                ] + ('normal ' + config.run.samples).split() + [
+                    '-V', config.genotype_snps.reference_version,
+                    '-b', f'{output}/baf/tumor.1bed',
+                    '-O', f'{output}/rdr'
+                ] + extra_args
+            )
+
+        # ----------------------------------------------------
+
+        if config.run.combine_counts:
+            _stdout = sys.stdout
+            sys.stdout = StringIO()
+            os.makedirs(f'{output}/abin', exist_ok=True)
+
+            phasefile = f'{output}/phase/phased.vcf.gz'
+            args=[
+                '-A', f'{output}/rdr',
+                '-b', f'{output}/baf/tumor.1bed',
+                '-t', f'{output}/rdr/total.tsv', 
+                '-V', config.genotype_snps.reference_version,
+                '-o', f'{output}/abin/bulk.bb'
+                ] + extra_args
+            
+            if os.path.exists(phasefile):
+                log(msg="Found phasing file, including phasing in binning process.\n", level = "INFO")
+                args = ['-p', f'{output}/phase/phased.vcf.gz'] + args
+            else:
+                log(msg=f"NO PHASING FILE FOUND at [phasefile]. Not including phasing in binning process.\n", level = "INFO")
+                
+            combine_counts(args)
+
+            #out = sys.stdout.getvalue()
```

Comment:
commented code

Reply:
I think I cleaned up a few of these. Added a note on this in the Python Developer Guidelines wiki.